tools/preflight/preflight.go

```diff
@@ -0,0 +1,620 @@
+	}
+
+	//  Wait for snapshot pod to become ready.
+	o.Logger.Infoln("Waiting for restore pod to become ready")
+	waitOptions.Name = restorePodSpec.GetName()
+	err = waitUntilPodCondition(o.Context, waitOptions)
+	if err != nil {
+		return err
+	}
+	o.Logger.Infoln(fmt.Sprintf("%s Created restore pod from volume snapshot", check))
+
+	execOp = exec.Options{
+		Namespace:     o.Namespace,
+		Command:       []string{"ls", "/demo/data/sample-file.txt"},
+		PodName:       restorePodSpec.GetName(),
+		ContainerName: busyboxContainerName,
+		Executor:      &exec.DefaultRemoteExecutor{},
+		Config:        restConfig,
+		ClientSet:     clientSet,
+	}
+	err = execInPod(&execOp, o.Logger)
+	if err != nil {
+		return err
+	}
+	o.Logger.Infoln("restored pod has expected data")
+
+	// TODO - add timeout for delete
+	o.Logger.Infoln(fmt.Sprintf("Deleting source pod - %s", srcPod.GetName()))
+	err = deleteK8sResourceWithForceTimeout(o.Context, srcPod, o.Logger)
+	if err != nil {
+		o.Logger.Warnln(fmt.Sprintf("problem occurred deleting source pod - %s :: %s", srcPod.GetName(), err.Error()))
+	}
+	o.Logger.Infoln(fmt.Sprintf("%s Deleted source pod", check))
+
+	unmountedVolSnapSrcSpec := createVolumeSnapsotSpec(unmountedVolumeSnapSrc+resNameSuffix, o.Namespace, snapshotVer, pvc.GetName())
+	if err = runtimeClient.Create(o.Context, unmountedVolSnapSrcSpec); err != nil {
+		o.Logger.Errorln(err.Error())
+		o.Logger.Errorln(fmt.Sprintf("%s error creating volume snapshot from unmounted source pvc", cross))
+		return err
+	}
+	o.Logger.Infoln("Waiting for Volume snapshot from source pvc to become 'readyToUse:true'")
+	err = waitUntilVolSnapReadyToUse(unmountedVolSnapSrcSpec, snapshotVer, getDefaultRetryBackoffParams())
+	if err != nil {
+		return err
+	}
+	o.Logger.Infoln(fmt.Sprintf("%s Created volume snapshot from source pvc and is ready-to-use", check))
+
+	// create unmounted restore pvc and pod
+	unmountedPvcSpec := createRestorePVCSpec(unmountedRestorePvc+resNameSuffix,
+		unmountedVolumeSnapSrc+resNameSuffix,
+		o.StorageClass, o.Namespace)
+	_, err = clientSet.CoreV1().PersistentVolumeClaims(o.Namespace).
+		Create(o.Context, unmountedPvcSpec, metav1.CreateOptions{})
+	if err != nil {
+		o.Logger.Errorln(err.Error())
+		return err
+	}
+	unmountedPodSpec := createRestorePodSpec(unmountedRestorePod+resNameSuffix, unmountedPvcSpec.GetName(), o)
+	unmountedPodSpec, err = clientSet.CoreV1().Pods(o.Namespace).Create(o.Context, unmountedPodSpec, metav1.CreateOptions{})
+	if err != nil {
+		o.Logger.Errorln(err.Error())
+		return err
+	}
+	// wait for unmounted restore pod to become ready
+	waitOptions.Name = unmountedPodSpec.GetName()
+	o.Logger.Infoln("Waiting for unmounted restore pod to become ready")
+	err = waitUntilPodCondition(o.Context, waitOptions)
+	if err != nil {
+		return err
+	}
+	o.Logger.Infoln("Created restore pod from volume snapshot of unmounted pv")
+
+	execOp.PodName = unmountedPodSpec.GetName()
+	execOp.ContainerName = busyboxContainerName
```

Comment:
not needed... same name is set already, check once

Reply:
Different name is set. Hence, need to change name here